test/test_type_promotion.py

```diff
@@ -118,6 +118,10 @@ def test_complex_scalar_mult_tensor_promotion(self, device):
         b = torch.tensor([2j, 2j], device=device)
         self.assertEqual(a, b)
         self.assertEqual(a.dtype, b.dtype)
+        a = a + torch.tensor([True, ])
+        b = b + 1
+        self.assertEqual(a, b)
+        self.assertEqual(a.dtype, b.dtype)
```

Comment:
nit: you can also add an `exact_dtype` arg to assertEquals to check the dtype (this was added recently I think).